Path to build system is wrong in generate_vcvrack script

The path to build system is wrong in the `generate_vcvrack.py` script.

Therefore, one cannot make changes to UI from within the IDE.

```
python3 artifacts/generate_vcvrack.py "KickCut24.erbui"
Traceback (most recent call last):
  File "artifacts/generate_vcvrack.py", line 20, in <module>
    import erbui
ModuleNotFoundError: No module named 'erbui'
make: *** [artifacts/deploy_vcvrack.py] Error 1
```

This is because the path is relative and hardcoded.

Workaround: call `configure.py` for each UI changes instead.

--- a/build-system/build-system.gyp
+++ b/build-system/build-system.gyp
@@ -20,7 +20,7 @@
 
             # erbb
             'erbb/__init__.py',
-            'erbb/generate_vcvrack.py',
+            'erbb/generate_vcvrack_template.py',
             'erbb/vscode/launch.json',
             'erbb/vscode/tasks.json',
 

--- a/build-system/erbb/__init__.py
+++ b/build-system/erbb/__init__.py
@@ -64,10 +64,28 @@ def configure_native (name, path):
          if 'BuildIndependentTargetsInParallel' in line:
             print ('\t\t\t\tLastUpgradeCheck = 1000;')
 
-   shutil.copyfile (
-      os.path.join (PATH_THIS, 'generate_vcvrack.py'),
-      os.path.join (path_artifacts, 'generate_vcvrack.py')
-   )
+   configure_native_vcvrack (path)
+
+
+
+"""
+==============================================================================
+Name: configure_native_vcvrack
+==============================================================================
+"""
+
+def configure_native_vcvrack (path):
+   path_artifacts = os.path.join (path, 'artifacts')
+   path_py = os.path.join (path_artifacts, 'generate_vcvrack.py')
+   path_template = os.path.join (PATH_THIS, 'generate_vcvrack_template.py')
+
+   with open (path_template, 'r') as file:
+      template = file.read ()
+
+   template = template.replace ('%PATH_ROOT%', PATH_ROOT)
+
+   with open (path_py, 'w') as file:
+      file.write (template)
 
 
 

--- a/build-system/erbb/generate_vcvrack_template.py
+++ b/build-system/erbb/generate_vcvrack_template.py
@@ -1,6 +1,6 @@
 #!/usr/bin/env python3
 #
-#     configure.py
+#     generate_vcvrack.py
 #     Copyright (c) 2020 Raphael Dinge
 #
 #Tab=3########################################################################
@@ -13,8 +13,7 @@ import os
 import subprocess
 import sys
 
-PATH_THIS = os.path.abspath (os.path.dirname (__file__))
-PATH_ROOT = os.path.abspath (os.path.dirname (os.path.dirname (os.path.dirname (PATH_THIS))))
+PATH_ROOT = '%PATH_ROOT%'
 
 sys.path.insert (0, os.path.join (PATH_ROOT, 'build-system'))
 import erbui

--- a/include/erb/erb.gypi
+++ b/include/erb/erb.gypi
@@ -212,7 +212,7 @@
                      '<!(echo artifacts/plugin_vcvrack.cpp)',
                      '<!(echo artifacts/plugin.json)',
                   ],
-                  'action': [ 'python3', 'artifacts/generate_vcvrack.py', '<(RULE_INPUT_PATH)' ],
+                  'action': [ '<!(which python3)', 'artifacts/generate_vcvrack.py', '<(RULE_INPUT_PATH)' ],
                },
             ],
 
@@ -247,7 +247,7 @@
                {
                   'postbuild_name': 'Copy to VCV Rack plug-ins folder',
                   'action': [
-                     'python3', 'artifacts/deploy_vcvrack.py'
+                     '<!(which python3)', 'artifacts/deploy_vcvrack.py'
                   ],
                },
             ],

